packages/deployments/contracts/contracts_forge/core/connext/facets/ProposedOwnableFacet.t.sol

```diff
@@ -132,6 +141,45 @@ contract ProposedOwnableFacetTest is ProposedOwnableFacet, FacetHelper {
     utils_acceptNewOwnerAndAssert(_proposed);
   }
 
+  function utils_assignRoleRouter(address routerAgent, address caller) public {
+    vm.expectEmit(true, true, true, true);
+    emit AssignRouterRole(routerAgent);
+    vm.prank(caller);
+    this.assignRoleRouter(routerAgent);
+
+    if (this.queryRole(routerAgent) == Role.Router) {
+      assertTrue(true);
+    } else {
+      assertTrue(false);
+    }
+  }
+
+  function utils_assignRoleWatcher(address watcherAgent, address caller) public {
+    vm.expectEmit(true, true, true, true);
+    emit AssignWatcherRole(watcherAgent);
+    vm.prank(caller);
+    this.assignRoleWatcher(watcherAgent);
+
+    if (this.queryRole(watcherAgent) == Role.Watcher) {
+      assertTrue(true);
+    } else {
+      assertTrue(false);
+    }
+  }
+
+  function utils_assignRoleAdmin(address adminAgent, address caller) public {
+    vm.expectEmit(true, true, true, true);
+    emit AssignAdminRole(adminAgent);
+    vm.prank(caller);
+    this.assignRoleAdmin(adminAgent);
+
+    if (this.queryRole(adminAgent) == Role.Admin) {
+      assertTrue(true);
+    } else {
+      assertTrue(false);
+    }
+  }
+
```

Comment:
Ideally should have tests for fail cases, as well as have some tests around the intended/desired behavior when a candidate address is switched to a different role from a previous one OR we attempt to assign a candidate to multiple roles (not sure what the desired behavior is out of the two of these...)

Reply:
we do have test for failed cases, will add more after making the above changes.